persistence/javadsl/src/main/scala/com/lightbend/lagom/internal/javadsl/persistence/ReadSideActor.scala

```diff
@@ -3,103 +3,108 @@
  */
 package com.lightbend.lagom.internal.javadsl.persistence
 
-import akka.{ Done, NotUsed }
-import akka.actor.{ Actor, ActorLogging, Props, Status }
+import akka.actor.{ Actor, ActorLogging, Props }
+import akka.cluster.sharding.ShardRegion.EntityId
 import akka.stream.javadsl.Source
-import akka.stream.scaladsl.{ Keep, Sink }
-import akka.stream.{ KillSwitch, KillSwitches, Materializer }
+import akka.stream.scaladsl.{ Keep, RestartSource, Sink }
+import akka.stream.{ KillSwitch, KillSwitches, Materializer, scaladsl }
 import akka.util.Timeout
+import akka.{ Done, NotUsed }
+import com.lightbend.lagom.internal.persistence.ReadSideConfig
 import com.lightbend.lagom.internal.persistence.cluster.ClusterDistribution.EnsureActive
 import com.lightbend.lagom.internal.persistence.cluster.ClusterStartupTask
 import com.lightbend.lagom.javadsl.persistence._
 
 import scala.compat.java8.FutureConverters._
-import scala.concurrent.duration.FiniteDuration
 
 private[lagom] object ReadSideActor {
 
   def props[Event <: AggregateEvent[Event]](
-    processor:            () => ReadSideProcessor[Event],
-    eventStreamFactory:   (AggregateEventTag[Event], Offset) => Source[akka.japi.Pair[Event, Offset], NotUsed],
-    clazz:                Class[Event],
-    globalPrepareTask:    ClusterStartupTask,
-    globalPrepareTimeout: FiniteDuration
+    config:             ReadSideConfig,
+    clazz:              Class[Event],
+    globalPrepareTask:  ClusterStartupTask,
+    eventStreamFactory: (AggregateEventTag[Event], Offset) => Source[akka.japi.Pair[Event, Offset], NotUsed],
+    processor:          () => ReadSideProcessor[Event]
   )(implicit mat: Materializer) = {
-    Props(classOf[ReadSideActor[Event]], processor, eventStreamFactory, clazz, globalPrepareTask, globalPrepareTimeout, mat)
+    Props(
+      classOf[ReadSideActor[Event]],
+      config,
+      clazz,
+      globalPrepareTask,
+      eventStreamFactory,
+      processor,
+      mat
+    )
   }
 
-  /**
-   * Start processing from the given offset
-   */
-  case class Start(offset: Offset)
+  case object Start
+
```

Comment:
it is not necessary to propagate the `offset` because the `RestartSource` include the invocation of `prepare` that obtains the offset and the startup of the source.